### PR TITLE
feat: improve grid validation 

### PR DESCRIPTION
**Pull request by dzbo**
### Ticket ID

[DEV-11849](https://app.clickup.com/t/2645698/DEV-11849)
[DEV-11908](https://app.clickup.com/t/2645698/DEV-11908)
[DEV-11916](https://app.clickup.com/t/2645698/DEV-11916)

### Description

- integrate zod for building validation schemas
- refactor validation and forms when adding new widgets
- improve validation and add purification to LSP28 config data

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-11849 Improve config validation for properties](https://app.clickup.com/t/2645698/DEV-11849)

**Comment by Hugoo** [comment]
Task linked: [DEV-11908 Flip widget resize icon to match resize direction](https://app.clickup.com/t/2645698/DEV-11908)

**Comment by Hugoo** [comment]
Task linked: [DEV-11916 Bug: if we leave edit mode on an empty grid, it shows nothing](https://app.clickup.com/t/2645698/DEV-11916)

File: domains/schema/transforms/platformTransform.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const parsedValues = await parsePlatformInput(type, values.src)
+    return parsedValues.properties
+  } catch (error: unknown) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: formatMessage('errors_invalid_input', {
+        name: capitalize(type),
+      }),
+      path: ['src'],
+    })
```
Comment (JeneaVranceanu) [inline domains/schema/transforms/platformTransform.ts:29]: Why don't we use `error` object in the `addIssue`? It should have the information about which field is invalid.

**Review by JeneaVranceanu — APPROVED** [review]

File: domains/schema/transforms/platformTransform.ts (hunk 1 above)
Comment (dzbo) [reply domains/schema/transforms/platformTransform.ts:29]: By `error` object you mean `ctx` instance? I've checked but it's empty in this case:

Comment (JeneaVranceanu) [reply domains/schema/transforms/platformTransform.ts:29]: No, the `error: unknown`. 
It should return the information about the invalid field if such error is thrown.

**Review by doubleppereira — APPROVED** [review]

File: domains/schema/transforms/platformTransform.ts (hunk 1 above)
Comment (dzbo) [reply domains/schema/transforms/platformTransform.ts:29]: The error we catch here comes from `parsePlatformInput`, it doesn't have anything to do with zod unfortunatelly :/

**Pull request merged**